fs/fs.go

```diff
@@ -81,13 +82,66 @@ var (
 	fusermountBin               = "fusermount"
 )
 
+type layerQueue struct {
+	mu       sync.Mutex
+	queue    chan func() string
+	cache    *hashset.Set
+	maxSize  int
+	currSize int
+}
+
+func (lq *layerQueue) Start() error {
+	if lq.queue != nil {
+		return fmt.Errorf("layer queue already instantiated")
+	}
+
+	lq.queue = make(chan func() string)
+	lq.cache = hashset.New()
+
+	go func() {
```

Comment:
nit: we should probably associate this go-routine with an errgroup or waitgroup so that it can be closed later. It should probably also select over a context so it can be canceled.

Reply:
Not super familiar with how exactly this works, does the new version this look about right?

Reply:
Do we want a wait/err group? If the current queue is finished resolving, than the `layerQueue` will exit as-well. Meaning, if you were to pull an image some time after an initial pull, than the `layerQueue` will be inactive and so all layer resolves will be sequential, right?

I think the `layerQueue` should be tied to the lifecycle of the process/filesystem.